### ISSUE TITLE
Jest encountered an unexpected token

### ISSUE THREAD

**Issue by EvgenyOrekhov** [ant-design/ant-design-pro#3366]
Steps to reproduce:

1. Clone the repo
2. `npm install`
3. `npm run test:all`

Expected result: all tests should pass.

Actual result: 1 test fails with "Jest encountered an unexpected token" error.

Output:

```text
> umi test "--maxWorkers=1" "--runInBand"

 PASS  src/components/Authorized/CheckPermissions.test.js (5.071s)
 PASS  src/utils/utils.test.js
 PASS  src/components/PageHeader/index.test.js
 PASS  src/e2e/login.e2e.js (5.769s)
 PASS  src/components/AvatarList/index.test.js
 PASS  src/e2e/baseLayout.e2e.js (10.054s)
 PASS  src/e2e/userLayout.e2e.js
 PASS  src/components/SiderMenu/SiderMenu.test.js
 PASS  src/utils/authority.test.js
 PASS  src/e2e/topMenu.e2e.js
 PASS  src/components/_utils/pathTools.test.js
 PASS  src/e2e/home.e2e.js
 PASS  src/components/Ellipsis/index.test.js
 FAIL  src/pages/Result/Success.test.js
  ● Test suite failed to run

    Jest encountered an unexpected token

    This usually means that you are trying to import a file which Jest cannot parse, e.g. it's not plain JavaScript.

    By default, if Jest sees a Babel config, it will use that to transform your files, ignoring "node_modules".

    Here's what you can do:
     • To have some of your "node_modules" files transformed, you can specify a custom "transformIgnorePatterns" in your config.
     • If you need a custom transformation specify a "transform" option in your config.
     • If you simply want to mock your non-JS modules (e.g. binary assets) you can stub them out with the "moduleNameMapper" config option.

    You'll find more details and examples of these config options in the docs:
    https://jestjs.io/docs/en/configuration.html

    Details:

    /home/user/ant-design-pro/node_modules/umi/lib/link.js:1
    ({"Object.<anonymous>":function(module,exports,require,__dirname,__filename,global,jest){import { Link } from 'react-router-dom';
                                                                                                    ^

    SyntaxError: Unexpected token {

      10 | const PageHeaderWrapper = ({ children, contentWidth, wrapperClassName, top, ...restProps }) => (
      11 |   <div style={{ margin: '-24px -24px 0' }} className={wrapperClassName}>
    > 12 |     {top}
         |          ^
      13 |     <MenuContext.Consumer>
      14 |       {value => (
      15 |         <PageHeader

      at ScriptTransformer._transformAndBuildScript (node_modules/jest-runtime/build/script_transformer.js:403:17)
      at Object.<anonymous> (src/components/PageHeaderWrapper/index.js:12:36)
      at Object.<anonymous> (src/pages/Result/Success.js:16:49)
      at Object.<anonymous> (src/pages/Result/Success.test.js:7:39)

Test Suites: 1 failed, 13 passed, 14 total
Tests:       57 passed, 57 total
Snapshots:   0 total
Time:        35.896s
Ran all test suites.
npm ERR! Test failed.  See above for more details.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! ant-design-pro@2.1.1 test:all: `node ./tests/run-tests.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the ant-design-pro@2.1.1 test:all script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/user/.npm/_logs/2019-01-11T13_52_23_674Z-debug.log
```

Ant Design Pro version: 7afe65c
Node.js version: 11.5.0
OS: Ubuntu 18.10

**Comment by imhele** [comment]
https://github.com/umijs/umi/issues/1821
umi@2.4.0 starts to use ESM instead of CommonJS, which jest does not support.

**Comment by sorrycc** [comment]
Fixed in umi@2.4.2
